src/components/GoodsList.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { get5First, getAll, getRedGoods } from '../api/goods';
+import './GoodsList.scss';
+
+type Props = {
+  getAll: () => Promise<Good[]>;
+  get5First: (goods: Good[]) => Good[];
+  getRedGoods: (goods: Good[]) => Good[];
+};
+type State = {
+  goods: Good[];
+  visibleGoods: Good[];
+};
+
+export class GoodsList extends React.Component<Props, State> {
+  state: State = {
+    goods: [],
+    visibleGoods: [],
+  };
+
+  componentDidMount() {
+    getAll()
+      .then(goods => {
+        this.setState({
+          goods,
+          visibleGoods: goods,
+        });
+      });
+  }
```

Comment:
getAll must be called after clicking on a button. You can create a new method that will work when you click the button